src/main/java/ladder/domain/LadderGame.java

```diff
@@ -1,19 +1,42 @@
 package ladder.domain;
 
 import ladder.dto.LadderGameRequest;
+import ladder.dto.LadderGameResult;
+import ladder.dto.LadderResultDto;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
 
 public class LadderGame {
 
-    private final LineList lineList = new LineList();
+    private final LadderBoard ladderBoard;
+
+    private final PlayerList participatingPlayerList;
 
-    private final PlayerList participatingPlayerList = new PlayerList();
+    private static final String ALL_PLAYERS = "all";
 
     public LadderGame(LadderGameRequest ladderGameRequest) {
-        participatingPlayerList.addBulk(ladderGameRequest.names());
-        lineList.addBulk(ladderGameRequest.height(), participatingPlayerList.size());
+        this.participatingPlayerList = PlayerList.of(ladderGameRequest.nameList());
+        this.ladderBoard = LadderBoard.of(ladderGameRequest);
+    }
+
+    public LadderResultDto ladderBoard() {
+        return new LadderResultDto(participatingPlayerList, ladderBoard);
+    }
+
+    public List<LadderGameResult> play(String playerName) {
+        if (playerName.equalsIgnoreCase(ALL_PLAYERS)) {
```

Comment:
playerName 이 null 인 경우 NPE 가 발생할 수 있는데요, 이러한 경우 반대로 값이 존재하는 것이 보장되는 레퍼런스를 기준으로 호출하면 null 케이스도 잡을 수 있습니다 :)
```suggestion
        if (ALL_PLAYERS.equalsIgnoreCase(playerName) {
```